src/main/webapp/app/exam/participate/exercises/exam-submission.component.ts

```diff
@@ -24,4 +27,8 @@ export abstract class ExamSubmissionComponent extends ExamPageComponent {
 
     abstract getSubmission(): Submission | undefined;
     abstract getExercise(): Exercise;
+    @Input() readonly = false;
+    @Input() examTimeline = false;
+    submissionVersion: SubmissionVersion;
```

Comment:
Can you make this `private`?

Reply:
no, because it's used in subclasses, but I made it protected.